plugins/module_utils/fusion.py

```diff
@@ -101,18 +102,25 @@ def get_fusion(module):
     }
 
     issuer_id = module.params[PARAM_ISSUER_ID]
-    key_file = module.params[PARAM_PRIVATE_KEY_FILE]
+    private_key_file = module.params[PARAM_PRIVATE_KEY_FILE]
+    private_key_password = module.params[PARAM_PRIVATE_KEY_PASSWORD]
+
+    if private_key_file is None and private_key_password is not None:
+        module.fail_on_missing_params([PARAM_PRIVATE_KEY_FILE])
```

Comment:
I think `private_key_file is None` is redundant, as `module.fail_on_missing_params` is defined to fail only when `private_key_file is None`.
```suggestion
    if private_key_password is not None:
        module.fail_on_missing_params([PARAM_PRIVATE_KEY_FILE])
```

Reply:
but `private_key_file` is not a required param by default. it becomes required only if `private_key_password` is set

Reply:
changed